cmd/scrape.go

```diff
@@ -0,0 +1,266 @@
+package cmd
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/gocolly/colly/v2"
+	"github.com/spf13/cobra"
+)
+
+var options             Options
+var url                 string
+
+type Article struct {
+	ID                  int             `json:"id"`
+    Title               string          `json:"title"`
+}
+
+type Site struct {
+    ID                  int             `json:"id"`
+    Url                 string          `json:"url"`
+	Selectors	        []string	    `json:"selectors"`
+    Articles            []Article       `json:"articles"`
+}
+
+type Options struct {
+    Sites               []Site          `json:"sites"`
+    Output              string          `json:"output"`
+    CurrentSite         Site            `json:"current_site"`
+}
+
+// This gets executed automatically when rootCmd.Execute() is called from main.go, due to it being a subcommand
+func init() {
+    rootCmd.AddCommand(scrapeCmd)
+	// Here are defined flags and configuration settings for commands
+    scrapeCmd.Flags().StringVarP(&options.Output, "output", "o", "./assets/news_site_articles.json", "set your json input/output file")
+	scrapeCmd.Flags().StringVarP(&url, "add-url", "a", "", "add new news source site url")
+}
+
+// Defining the scrape subcommand for use in CLI
+var scrapeCmd = &cobra.Command{
+	Use:   "scrape",
+	Short: "Command used for running news scraping scripts. The command will scrape all known sites if no flags are provided!",
+    Args: cobra.MinimumNArgs(0),
+	Run: func(cmd *cobra.Command, args []string) {
+        // Initialize options and run
+        options = Options{options.ReadFile(), options.Output, Site{}}
+        if url != "" {
+            exists, page := options.DoesPageExist(url)
+            if exists {
+                fmt.Println("The news source already exists!")
+                options.CurrentSite = page
+                page = page.Scrape()
```

Comment:
If you change the Scrape method to a pointer receiver as suggested above, then you will be able to change it to void and just turn these lines to `page.Scrape()`.